WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m

```diff
@@ -0,0 +1,489 @@
+#import "WPAnalyticsTrackerAutomatticTracks.h"
+#import <TracksService.h>
+#import "ContextManager.h"
+#import "AccountService.h"
+#import "BlogService.h"
+#import "WPAccount.h"
+#import "Blog.h"
+
+@interface WPAnalyticsTrackerAutomatticTracks ()
+
+@property (nonatomic, strong) TracksService *tracksService;
+@property (nonatomic, strong) NSDictionary *userProperties;
+@property (nonatomic, strong) NSString *anonymousID;
+
+@end
+
+@implementation WPAnalyticsTrackerAutomatticTracks
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _tracksService = [TracksService new];
+    }
+    return self;
+}
+
+- (void)track:(WPAnalyticsStat)stat
+{
+    [self track:stat withProperties:nil];
+}
+
+- (void)track:(WPAnalyticsStat)stat withProperties:(NSDictionary *)properties
+{
+    NSString *eventName = [self eventNameForStat:stat];
+    
+    if (eventName.length == 0) {
+        return;
+    }
+    
+    [self.tracksService trackEventName:eventName withCustomProperties:properties];
+}
+
+- (void)beginSession
+{
+    NSString *anonymousID = [[NSUserDefaults standardUserDefaults] stringForKey:@"TracksAnonymousUserID"];
+    if (!anonymousID) {
+        anonymousID = [[NSUUID UUID] UUIDString];
+        [[NSUserDefaults standardUserDefaults] setObject:anonymousID forKey:@"TracksAnonymousUserID"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+    
+    self.anonymousID = anonymousID;
+    
+    [self refreshMetadata];
+}
+
+- (void)endSession
+{
+    
+}
+
+- (void)refreshMetadata
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
```

Comment:
Perhaps we could reduce the duplication with the code over https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m#L84? Maybe some kind of class to generate this dictionary for us?